### PR TITLE
List: Reduce default space

### PR DESCRIPTION
**Pull request by felixhabib**
Reduce the default space between list items from `medium` to `small` or `xsmall` if the `size` prop is set to `small` or `xsmall`.

### CODE REVIEW

File: packages/braid-design-system/src/lib/components/List/List.tsx [hunk 1]
```diff
@@ -108,6 +108,10 @@ export const List = ({
     size: sizeProp,
     tone: toneProp,
   });
+
+  const defaultSpace = size === 'xsmall' ? 'xsmall' : 'small';
```
Comment (michaeltaranto) [inline packages/braid-design-system/src/lib/components/List/List.tsx:112]: Should also reduce the default for `small` text:

```suggestion
  const defaultSpace =
    size === 'xsmall' || size === 'small' ? 'xsmall' : 'small';
```

File: packages/braid-design-system/src/lib/components/List/List.tsx [hunk 2]
```diff
@@ -108,6 +108,10 @@ export const List = ({
...
+  const space = spaceProp ? spaceProp : defaultSpace;
```
Comment (michaeltaranto) [inline packages/braid-design-system/src/lib/components/List/List.tsx:113]:
```suggestion
  const space = spaceProp || defaultSpace;
```

File: .changeset/witty-donuts-sort.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+---
+
+---
+updated:
+  - List
+---
+
+**List**: Reduce default space between list items
+
+Reduce the default space between list items from `medium` to `small` or `xsmall` if the `size` prop is set to `xsmall`.
```
Comment (michaeltaranto) [inline .changeset/witty-donuts-sort.md:12]:
```suggestion
Reduce the default space between list items from `medium` to `small` or `xsmall` if the `size` prop is set to either `small` or `xsmall`.
```

File: packages/braid-design-system/src/lib/components/List/List.tsx (hunk 1 above)
Comment (felixhabib) [reply packages/braid-design-system/src/lib/components/List/List.tsx:112]: ahh, will update

File: packages/braid-design-system/src/lib/components/List/List.docs.tsx [hunk 4]
```diff
@@ -125,7 +125,7 @@ const docs: ComponentDocs = {
         source(
           <Columns space="large" collapseBelow="desktop">
             <Column>
-              <List tone="secondary" size="large" space="gutter">
+              <List tone="secondary" size="large" space="large">
```
Comment (michaeltaranto) [inline packages/braid-design-system/src/lib/components/List/List.docs.tsx:128]: I suggest removing all the `space` props from this section of the docs. That way its more about the text prop features, and demos the default spacing.

File: packages/braid-design-system/src/lib/components/List/List.docs.tsx [hunk 5]
```diff
@@ -113,40 +113,85 @@ const docs: ComponentDocs = {
...
+    },
+    {
+      label: 'Spacing',
+      description: (
+        <Text>
+          By default, Lists will use <Strong>small</Strong> spacing, or{' '}
+          <Strong>xsmall</Strong> spacing if the <Strong>size</Strong> prop is
+          set to either <Strong>small</Strong> or <Strong>xsmall</Strong>. This
+          can be changed via the <Strong>space</Strong> prop, which supports the
+          same the same spacing as{' '}
```
Comment (michaeltaranto) [inline packages/braid-design-system/src/lib/components/List/List.docs.tsx:165]:
```suggestion
          same spacing as{' '}
```

**Review by michaeltaranto — APPROVED** [review]

**Pull request merged**